extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultRecorder.java

```diff
@@ -0,0 +1,26 @@
+package io.quarkus.vault.runtime;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.vault.runtime.config.VaultServerRuntimeConfig;
+
+@Recorder
+public class VaultRecorder {
+
+    private static final Logger log = Logger.getLogger(VaultRecorder.class);
+
+    public static final String REST_URL = "quarkus.vault.url";
+
+    public void configureRuntimeProperties(VaultServerRuntimeConfig vaultServerRuntimeConfig) {
+
+        if (ConfigProvider.getConfig().getOptionalValue(REST_URL, String.class).isPresent()) {
```

Comment:
Why do you do that instead of using `vaultServerRuntimeConfig`?

Reply:
I think some very old code, from when I was trying to use the jax-rs client. not sure. anyway I used your suggestion.